### ISSUE TITLE
unused-value linter should catch more cases

### ISSUE THREAD

**Issue by orestis** [clj-kondo/clj-kondo#1899]
**version**

Calva version used: v2.0.321
clojure-lsp version used: 2022.11.03-00.16.44-nightly
clj-kondo version used: 2022.11.02

**platform**

Whatever Calva uses, I think native-image?

**editor**

VSCode/Calva

**problem**

```
(when-let [x true]
  x
  "something else")
```

Since `x` is an unused value, I would expect it to be reported as such. But I think, that since it appears twice in the block, there's something there to block it. If I use a literal, then it gets reported as unused value.

**repro**

```
(ns repro)

(when-let [x true]
  x
  "something else")
```

```
> clj-kondo --lint repro.clj
linting took 20ms, errors: 0, warnings: 0
```

**config**

No config.

**expected behavior**

I would expect that `x` is reported as an unused value, since it's not in the final position of the `when` form.

**Comment by borkdude** [comment]
Since a local reference is always "pure" I think it's say to conclude it's always an unused value when not in tail position in `when` or wherever.

**Comment by borkdude** [comment]
You're right about that, any reference, good point :)
